sql-style-guide.md

```diff
@@ -0,0 +1,124 @@
+# SQL Style Guide
+
+This guide establishes the standards for SQL in Boundary.
```

Comment:
perhaps `SQL migrations`

Reply:
The goal is that it will eventually cover all SQL.